skimage/filters/thresholding.py

```diff
@@ -530,6 +531,21 @@ def threshold_li(image, *, tolerance=None):
         is less than this value. By default, this is half the smallest
         difference between intensity values in ``image``.
 
+    initial_guess : float or Callable[[array[float]], float], optional
```

Comment:
This is so complicated. I know you want to make a generic function, but you could just call your callable before calling `threshold_li`

IMO, initial_guess should just be `float`